### ISSUE TITLE
Add an oEmbed Content Provider

### ISSUE THREAD

**Issue by drub0y** [JabbR/JabbR#487]
[oEmbed](http://oembed.com/) is a relatively simple standard that many content providers already support (youtube, soundcloud, etc.). It would almost be best to implement this as the "catch-all" provider in the case that no specific content provider implementation steps up to handle a specific base URL.

**Comment by davidfowl** [comment]
As long as it supports https.

**Comment by drub0y** [comment]
Yeah, well as you've seen that's going to be a challenge... not unique to this. We'll have to figure out what's possible. 

I had done some more research and [there's an oembed jQuery plug-in](http://code.google.com/p/jquery-oembed/) if we wanted to do it all client side, but then the provider would have to offer HTTPS itself.

Then there's [Noembed](http://www.noembed.com/) which apparently offers an HTTPS JSONP proxy to many popular sites.

Finally the big player seems to be [Embedly](http://embed.ly/) and they [apparently are beta'ing some kind of version of their API that supports https](http://support.embed.ly/entries/20250727-version-of-embed-ly-appropriate-for-an-https-site). Of course they cost money for their solution.
